source/ui/input-text-model.cpp

```diff
@@ -1,43 +1,66 @@
-#include <iostream>
+/*
+  * C++ Assembler Interpreter
+  * Copyright (C) 2016 Chair of Computer Architecture
+  * at Technical University of Munich
+  *
+  * This program is free software: you can redistribute it and/or modify
+  * it under the terms of the GNU General Public License as published by
+  * the Free Software Foundation, either version 3 of the License, or
+  * (at your option) any later version.
+  *
+  * This program is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program. If not, see <http://www.gnu.org/licenses/>.
+  */
 
 #include "core/conversions.hpp"
 #include "ui/input-text-model.hpp"
 
 InputTextModel::InputTextModel(QQmlContext* context, MemoryAccess memoryAccess)
 : QObject()
-, context(context)
-, start(0)
-, maxLength(20)
-, memoryAccess(memoryAccess) {
-  context->setContextProperty("inputtextMod", this);
+, _context(context)
+, _start(0)
+, _maximumLength(20)
+, _memoryAccess(memoryAccess) {
+  _context->setContextProperty("inputtextMod", this);
 }
 
 void InputTextModel::newText(QString text) {
   std::string stdText = text.toStdString();
   for (size_t i = 0; i < stdText.length(); i++) {
+    if(_start + i > _memoryAccess.getMemorySize().get()){
+        //To long
+        break;
+    }
     char z = stdText.at(i);
```

Comment:
This doesn't really make sense. `unsigned != int`. Also you should use `static_cast` to perform casts. But here, I don't even see why you need to cast at all:
```
auto number = stdText[i]
```